### PR TITLE
Restore codegen tests

### PR DESCRIPTION
**Pull request by matanlurey**
By removing `reflection_remover` (???) and doing the steps manually instead.

(/cc @natebosch)

Closes https://github.com/dart-lang/angular2/issues/272

### CODE REVIEW

**Comment by natebosch** [comment]
Looks like the tests time out on Travis?

**Comment by matanlurey** [comment]
Runs fine locally. Will need some @kevmoo magic.

**Comment by kevmoo** [comment]
This breaks on my machine w/ timeouts:

`TEST_PLATFORM=content-shell ./tool/travis.sh`

```
pub run test -x codegen -p content-shell
05:00 +1182 -1: loading test/common/directives/if_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:00 +1182 -2: loading test/common/directives/ng_template_outlet_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:00 +1182 -3: loading test/common/forms/accessor_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:26 +1182 -4: loading test/compiler/preserve_whitespace_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:37 +1182 -5: loading test/core/change_detection/component_state_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:38 +1182 -6: loading test/core/debug/debug_node_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:43 +1182 -7: loading test/core/directive_lifecycle_integration_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage

05:50 +1182 -8: loading test/core/linker/component_selector_test.dart [E]
  TimeoutException after 0:05:00.000000: Test timed out after 5 minutes.
  dart:isolate  _RawReceivePortImpl._handleMessage
```

**Comment by kevmoo** [comment]
Ditto `TEST_PLATFORM=firefox ./tool/travis.sh`

```
Error: Can't read 'file:///Users/kevmoo/source/github/angular2/test/common/directives/if_test.template.dart' (Error reading 'test/common/directives/if_test.template.dart'  (No such file or directory)).
import 'if_test.template.dart' as ng_codegen;
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by kevmoo** [comment]
@matanlurey more thoughts on this?

**Comment by matanlurey** [comment]
I ended up spending too much time on this with no real results.

tl;dr, and maybe @grouma is needed - excluding or including tags doesn't effect what is loaded in the browser, so it crashes when it tries to load AOT tests in a non-AOT environment or vica versa.

**Comment by kevmoo** [comment]
Maybe use `paths` - https://github.com/dart-lang/test/blob/master/doc/configuration.md#paths

in the config to include/exclude files for codegen?

**Comment by matanlurey** [comment]
That doesn't seem to mesh well with our structure.
